proto/transaction.proto

```diff
@@ -26,6 +26,7 @@ message TransactionInternal {
     map<string, MasterMetadata> master_metadata = 2;
     TransactionType type = 3;
     internal.MachineId coordinating_server = 4;
+    uint32 replica_id = 5;
```

Comment:
is this still necessary given that we're getting the `replica_id` from the metadata?

Reply:
It's needed in the scheduler also to deal with the remaster manager. We could calculate it on the fly but it seems nice to have it along with the regular id

Reply:
I might miss something here. Is `replica_id` the home of the current transaction? If it is the case then it is not applicable for multi-home txn, and for single-home txn, you can get that same information from `master_metadata` which is in the same struct.

Reply:
This is the TxnReplicaId. Renaming it to that is a good idea